src/tuner/nccl_ofi_model.c

```diff
@@ -1,123 +1,260 @@
-#include "config.h"
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
 
-#include <stdlib.h>
-#include <math.h>
+#include "config.h"
+#include <stdbool.h>
 
 #include "nccl-headers/nvidia/tuner.h"
 #include "nccl_ofi_tuner.h"
-#include "nccl_ofi_log.h"
 #include "nccl_ofi_math.h"
 
-float nccl_ofi_tuner_compute_base_cost(ncclFunc_t func, int algo, int proto)
+#include <float.h>
+#include <math.h>
+
+#define GIBI                                 (1.0 * 1024ULL * 1024ULL * 1024ULL)
+#define MICRO                                (1e-6)
+
+/* EFA unidirectional network bandwidth */
+#define NCCL_OFI_TUNER_NET_NUM_RAILS         (4.0)  /* Available to each GPU */
+#define NCCL_OFI_TUNER_INTERNODE_BW_PER_RAIL (12.5) /* per rail */
+#define NCCL_OFI_TUNER_INTERNODE_BW                                                   \
+	(NCCL_OFI_TUNER_NET_NUM_RAILS * NCCL_OFI_TUNER_INTERNODE_BW_PER_RAIL * GIBI * \
+	 MICRO) /* per accelerator */
```

Comment:
Comment on units

Reply:
unchanged from before, gibi per microsecond

Reply:
Just looking at these defs, the units aren't obvious to me, so I think a comment would help make things clear.

Suggest:
```
#define NCCL_OFI_TUNER_INTERNODE_BW_PER_RAIL (12.5) /* GB/s, per rail */
#define NCCL_OFI_TUNER_INTERNODE_BW ... /* B/usec, per accelerator */
```

Reply:
Also,
```
/* EFA unidirectional network bandwidth */
```
These are bidirectional bandwidths (i.e., each EFA rail is 100 Gbps in each direction); I would update this comment accordingly (unidirectional => bidirectional).

Reply:
There's a lot more we have to do here, I've added this to a notes.txt file I maintain, I'll make sure we take care of it on the next patchset.